[native] Add Mutual TLS support to Arrow Flight connector (#25388)
prestodb/presto: presto-docs/src/main/sphinx/connector/base-arrow-flight.rst

``arrow-flight.server-ssl-enabled``         Port is ssl enabled
========================================== ==============================================================

.. note::

   A secure TLS connection is enabled whenever the server certificate
   (``arrow-flight.server-ssl-certificate``) is provided, even if the
   ``arrow-flight.server-ssl-enabled`` flag is not set.

   If a client certificate (``arrow-flight.client-ssl-certificate``) is also provided,
   the Flight client will be configured to perform mutual authentication.

>    However, a true mutual TLS (mTLS) connection is only established if the Flight server
>    is explicitly configured to **require and verify** client certificates.

Reviewer: ```suggestion
   Note that the Flight server must also be properly configured to **require and verify** client certificates to
   enable mutual TLS (mTLS).
```

> 

Reviewer: It would also be good to show a code snippet to set these options in the flight server
